markdown: Clean up markdown_help.html with frontend markdown processor

Signior-X opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
This is a follow-up PR for #16891 
Done the changes as suggested by @timabbott in the review.

Checklist for changes requested:
- [x] Rebased and solved conflicts
- [x] Used a more direct selector instead of next
- [x] Removed empty `td` tags
- [x] Used the frontend parser wherever possible

Fixes #15375 
**Testing plan:** <!-- How have you tested? -->
Tested manually and still checking in progress.

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
@Signior-X thanks for working on this!  However, it doesn't seem to work correctly -- if you compare with chat.zulip.org, the formatting is a bit messed up.  Also we should be able to do things like this as well:


```
diff --git a/templates/zerver/app/markdown_help.html b/templates/zerver/app/markdown_help.html
index 9cd6c67a08..a1e410bcf9 100644
--- a/templates/zerver/app/markdown_help.html
+++ b/templates/zerver/app/markdown_help.html
@@ -27,36 +27,19 @@
                         <td class="rendered_markdown"><a href="https://zulip.org" target="_blank" rel="noopener noreferrer">Zulip website</a></td>
                     </tr>
                     <tr>
-                        <td>* Milk<br>
+                        <td class="apply_markdown">* Milk<br>
                             * Tea<br>
                             &nbsp;&nbsp;* Green tea<br>
                             &nbsp;&nbsp;* Black tea<br>
                             &nbsp;&nbsp;* Oolong tea<br>
                             * Coffee
                         </td>
-                        <td class="rendered_markdown">
-                            <ul>
-                                <li>Milk</li>
-                                <li>Tea
-                                    <ul>
-                                        <li>Green tea</li>
-                                        <li>Black tea</li>
-                                        <li>Oolong tea</li>
-                                    </ul>
-                                </li>
-                                <li>Coffee</li>
-                            </ul>
-                        </td>
                     </tr>
                     <tr>
```

Also, please squash the commits -- the first has a merge conflict that is cleaned up in the second.

Signior-X commented:
@timabbott thanks for the review. I will update the PR accordingly.

Signior-X commented:
@timabbott Done the changes as suggested.
- [x] Fixed the formatting
- [x] Used the frontend parser wherever possible, now also works for the list
- [x] Squashed the commits

However, the frontend parser was giving certain differences,
1. The arrow in the spoiler block is not present that is coming from the frontend markdown parser. It is working so no problem as such and used it.
2. The frontend parser was not giving html with syntax highlighting for code even using python, so not used the frontend parser for this case.

Rest almost every block is cleaned up.

timabbott commented:
Thanks for fixing those items.  I think it's probably problematic that we're not calling the code in `rendered_markdown.js` on these objects -- that's likely the cause for the issue with spoilers.

I rebased this and also did some code cleanup; this now fails our security linters.  I'll have a quick chat with @andersk on how to resolve that.  Once that's done, I think there's more we can do to cover more of our examples, but it'll require a bit of care.

ligmitz commented:
Using .append() instead of .html() passes the test.

timabbott commented:
> Using .append() instead of .html() passes the test.

Yes, but that only passes the test because it's avoiding the lint check, which only checks for `.html()` :).  @andersk perhaps we should add `.append` to the list for that linter?  

After some discussion with @andersk, we decided to do the next natural stage of this PR, namely turning this into a frontend template that would be better able to handle all of the pieces, which was just merged today as #17560.

Thanks for working on this @Signior-X!  Your work on this PR was essential to our figuring out that was the right path forward.

Signior-X commented:
Thanks @timabbott 
Nice to see, how this issue got solved!

andersk commented:
Opened #17581 for linting `.append`.
